docs/se/guides/40_webclient.adoc

```diff
@@ -0,0 +1,420 @@
+the web client activity. There are two ways to set up metrics, first one is programmatically on the WebClient instance,
+second one is to use configuration file.
+
+==== Add metrics dependency
+
+There is a specific dependency to use web client metrics in your application.
+
+[source,xml]
+.Add the following dependency to pom.xml:
+----
+<dependency>
+    <groupId>io.helidon.webclient</groupId>
+    <artifactId>helidon-webclient-metrics</artifactId>
+</dependency>
+----
+
+==== Set up metrics on WebClient instance
+
+It is possible to register metrics on WebClient directly into the code. Will be described a general method to use with
+any metrics.
+
+[source,java]
+.Example of metric creation:
+----
+import io.helidon.common.http.Http;
+import io.helidon.metrics.RegistryFactory;
+import io.helidon.webclient.metrics.WebClientMetrics;
+import io.helidon.webclient.spi.WebClientService;
+import org.eclipse.microprofile.metrics.MetricRegistry;
+import org.eclipse.microprofile.metrics|.Counter;
+                                       |.Meter;
+                                       |.Timer;
+                                       |.ConcurrentGauge;
+
+...
+
+public static void main(String[] args) {
+
+    MetricRegistry metricFactory = RegistryFactory.getInstance()
+            .getRegistry(MetricRegistry.Type.APPLICATION);
+
+    String metricName = "metric.GET.localhost";             // <1>
+
+    Counter counter = metricFactory|.counter(metricName);   // <2>
+                                   |.meter(metricName)
+                                   |.timer(metricName)
+                                   |.concurrentGauge(metricName)
+
+    WebClientService clientServiceMetric = WebClientMetrics|.counter()
+                                                           |.meter()
+                                                           |.timer()
+                                                           |.gaugeInProgress()
+
+                    .methods(Http.Method.GET)           // OPTIONAL
+                    .success(true)                      // OPTIONAL
+                    .errors(true)                       // OPTIONAL
+                    .description("Metric Description")  // OPTIONAL
+                    .nameFormat("counter.%1$s.%2$s")
+                    .build();                           // <3>
+----
+<1> Choose the metric name.
+<2> Create a metric from `metricFactory`.
+<3> Build a Web Client Service for counting the GET requests.
+
+The metric name can indicate what is measured. In this example, the metric target GET requests on the localhost.
+In order to pass this information to the webclient, the `nameFormat` method extracts it from the metric name.
+Otherwise, the metric name can also have nothing in common with its job. In this case, the methods with `OPTIONAL`
+comment are not required to be used. The `methods` will target the Http request type of your choice. `success` and `error` will
```

Comment:
The `methods` will target the chosen HTTP request type. while `success`and `error` will respectively measure if a request is successful or failed. The `description` will add a metric description.